### PR TITLE
Check if roms have already been installed before attempting to reinstall

### PR DESCRIPTION
**Pull request by jjshoots**

### CODE REVIEW

File: src/AutoROM.py [hunk 1]
```diff
@@ -197,6 +197,23 @@ def torrent_tar_to_buffer():
 
     return buffer
 
+def verify_installation(package, checksum_keys):
```
Comment (pseudo-rnd-thoughts) [inline src/AutoROM.py:200]: Add type hints

File: src/AutoROM.py [hunk 2]
```diff
@@ -197,6 +197,23 @@ def torrent_tar_to_buffer():
...
+
+        rom_path = os.path.join(package, file)
+        hash = hashlib.md5(open(rom_path,'rb').read()).hexdigest()
+
+        if not hash in checksum_keys:
+            return False
+
+        checksum_keys.remove(hash)
+
+    if len(checksum_keys) > 0:
```
Comment (pseudo-rnd-thoughts) [inline src/AutoROM.py:213]: `return len(check_sum_keys) == 0`

File: src/AutoROM.py [hunk 3]
```diff
@@ -197,6 +197,23 @@ def torrent_tar_to_buffer():
...
+        if not file.endswith(".bin"):
+            continue
+
+        rom_path = os.path.join(package, file)
+        hash = hashlib.md5(open(rom_path,'rb').read()).hexdigest()
+
+        if not hash in checksum_keys:
+            return False
+
+        checksum_keys.remove(hash)
```
Comment (pseudo-rnd-thoughts) [inline src/AutoROM.py:211]: As `checksum_keys` is a reference then we are modifying this reference, are we sure this doesn't mess something else up?

**Review by pseudo-rnd-thoughts — CHANGES_REQUESTED** [review]

File: src/AutoROM.py (hunk 3 above)
Comment (jjshoots) [reply src/AutoROM.py:211]: [Keys returns a copy](https://python-reference.readthedocs.io/en/latest/docs/dict/keys.html)

File: src/AutoROM.py (hunk 2 above)
Comment (jjshoots) [reply src/AutoROM.py:213]: done.

File: src/AutoROM.py (hunk 1 above)
Comment (jjshoots) [reply src/AutoROM.py:200]: Is this necessary? Everything else doesn't have typehints and it's not something that users are expected to use.

**Review by pseudo-rnd-thoughts — APPROVED** [review]

**Pull request merged**